cmd/abapEnvironmentRunAUnitTest.go

```diff
@@ -0,0 +1,569 @@
+		if x.Progress.Status == "Not Created" {
+			return "", err
+		}
+		if x.Progress.Status == "Completed" || x.Progress.Status == "FINISHED" {
+			return x.Link.Href, err
+		}
+		if x.Progress.Status == "" {
+			return "", fmt.Errorf("Could not get any response from AUnit poll: %w", errors.New("Status from AUnit run is empty. Either it's not an ABAP system or AUnit run hasn't started"))
+		}
+		time.Sleep(10 * time.Second)
+	}
+}
+
+func getHTTPResponseAUnitRun(requestType string, details abaputils.ConnectionDetailsHTTP, body []byte, client piperhttp.Sender) (*http.Response, error) {
+
+	log.Entry().WithField("ABAP Endpoint: ", details.URL).Info("Polling AUnit run status")
+
+	header := make(map[string][]string)
+	header["Accept"] = []string{"application/vnd.sap.adt.api.abapunit.run-status.v1+xml"}
+
+	req, err := client.SendRequest(requestType, details.URL, bytes.NewBuffer(body), header, nil)
+	if err != nil {
+		return req, fmt.Errorf("Getting AUnit run status failed: %w", err)
+	}
+	return req, err
+}
+
+func getResultAUnitRun(requestType string, details abaputils.ConnectionDetailsHTTP, body []byte, client piperhttp.Sender) (*http.Response, error) {
+
+	log.Entry().WithField("ABAP Endpoint: ", details.URL).Info("Getting AUnit results")
+
+	header := make(map[string][]string)
+	header["x-csrf-token"] = []string{details.XCsrfToken}
+	header["Accept"] = []string{"application/vnd.sap.adt.api.junit.run-result.v1+xml"}
+
+	req, err := client.SendRequest(requestType, details.URL, bytes.NewBuffer(body), header, nil)
+	if err != nil {
+		return req, fmt.Errorf("Getting AUnit run results failed: %w", err)
+	}
+	return req, err
+}
+
+func parseAUnitResult(body []byte, aunitResultFileName string) (err error) {
+	if len(body) == 0 {
+		return fmt.Errorf("Parsing AUnit result failed: %w", errors.New("Body is empty, can't parse empty body"))
+	}
+
+	responseBody := string(body)
+	log.Entry().Debugf("Response body: %s", responseBody)
+
+	//Optional checks before writing the Results
+	parsedXML := new(AUnitResult)
+	xml.Unmarshal([]byte(body), &parsedXML)
+	if len(parsedXML.Testsuite.Testcase) == 0 {
+		log.Entry().Info("There were no AUnit findings from this run")
+		return nil
+	}
+
+	//Write Results
+	err = ioutil.WriteFile(aunitResultFileName, body, 0644)
+	if err == nil {
+		log.Entry().Infof("Writing %s file was successful. Please find the results from the respective AUnit run in the %s file or in below logs", aunitResultFileName, aunitResultFileName)
+		//Logging of findings
+		log.Entry().Infof(`Here are the results for the AUnit test run '%s' executed by User %s on System %s in Client %s at %s. The AUnit run took %s seconds and contains %s tests with %s failures, %s errors, %s skipped and %s assert findings`, parsedXML.Title, parsedXML.System, parsedXML.ExecutedBy, parsedXML.Client, parsedXML.Timestamp, parsedXML.Time, parsedXML.Tests, parsedXML.Failures, parsedXML.Errors, parsedXML.Skipped, parsedXML.Asserts)
```

Comment:
Adapt text to account for the log level change

Reply:
This Info log summarizes the AUnit test runs and prints the information in to the log. The following logs will be leveled on Debug level.